apps/studio/components/interfaces/Storage/CreateBucketModal.tsx

```diff
@@ -95,12 +95,26 @@ const formId = 'create-storage-bucket-form'
 
 export type CreateBucketForm = z.infer<typeof FormSchema>
 
-export const CreateBucketModal = () => {
+interface CreateBucketModalProps {
+  hideAnalyticsOption?: boolean
+  visible?: boolean
+  onOpenChange?: (open: boolean) => void
```

Comment:
hmm im on the fence with introducing an external state like this - since there will (?) always be a button trigger to open this modal?